### ISSUE TITLE
Problem MOSEK with Power Cone and SDP

### ISSUE THREAD

**Issue by dcajasn** [cvxpy/cvxpy#2128]
**Describe the bug**
When I try to solve a problem that involves an SDP constraint and Power Cone constraint with MOSEK an error message "Error: Tyring to append a too big cone." appears but when I choose SCS I don't have problems.

**To Reproduce**
```python
import numpy as np
import pandas as pd
import cvxpy as cp
import riskfolio as rp
from scipy.linalg import sqrtm

rs = np.random.RandomState(seed=123)

N = 10
T = 1000
rows = int(N*(N+1)/2)

R = rs.randn(T, N)

x = cp.Variable((N,1))
X = cp.Variable((N,N), PSD=True)
psi = cp.Variable(nonneg=True)
gamma = cp.Variable((rows, 1))

D_2 = rp.duplication_matrix(N)
L_2 = rp.duplication_elimination_matrix(N)
S_2 = rp.duplication_summation_matrix(N)

Sigma_4 = rp.cokurtosis_matrix(R)
Sigma_4 = S_2 @ Sigma_4 @ S_2.T
Sqrt_Sigma_4 = sqrtm(Sigma_4)

y = L_2 @ cp.reshape(cp.vec(X), (N*N,1))
z = Sigma_4 @ y

M1 = cp.vstack([X, x.T])
M2 = cp.vstack([x, np.ones((1, 1))])
M3 = cp.hstack([M1, M2])

constraints = [cp.SOC(psi, Sqrt_Sigma_4 @ y),
               cp.sum(x) == 1,
               x >= 0,
               M3 >> 0,
               ]

constraints += [cp.constraints.power.PowCone3D(y, z, gamma, 0.5)]

obj = cp.pnorm(psi - gamma, p=2)

objective = cp.Minimize(obj)
problem = cp.Problem(objective, constraints)
problem.solve(solver='MOSEK')
# problem.solve(solver='SCS') # with this line works
```

**Expected behavior**
That the problem can be solve and return a x.value.

**Output**
<img width="762" alt="image" src="https://github.com/cvxpy/cvxpy/assets/61527316/5464b51b-0ff5-4a61-8b75-a98b54c81bfc">

**Version**
 - OS: macOS Ventura 13.3.1 
 - CVXPY Version: 1.3.1

**Additional context**
I use a M1 computer.

**Comment by aszekMosek** [comment]
Mosek is being instructed to add a cone where the variable indices exceed the number of variables so either 1) there is something wrong in the Mosek interface or 2) you are doing something illegal by importing the PowCone3D directly and bypassing the public interface or 3) some combination of 1 and 2. My knowledge and ability to decide stops here.

In any case replacing it with 

```
constraints += [cp.geo_mean(cp.vstack([y[i],z[i]]),[0.5,0.5])>=gamma[i] for i in range(rows)]
```

works well.

If you ever only want to use the 0.5 exponent then use the (rotated) quadratic cone or ``quad_over_lin`` instead of the general power cone.

**Comment by SteveDiamond** [comment]
This is definitely a bug on the cvxpy side. It's clear that the problem being constructed for Mosek is not right. I spent some time looking into it but I can't figure out what's happening to cause the error.

**Comment by dcajasn** [comment]
Thanks @SteveDiamond, I hope it could be solve for version 1.4 or 1.5 😅. By the moment I'm going to use the SOC constraint. I noticed this bug when I was trying to maximize a return/Relativistic Value at Risk ratio with a kurtosis constraint using Riskfolio-Lib, so I think this is not an issue that appears to most cvxpy users.

**Comment by rileyjmurray** [comment]
Resolved by #2131
